.github/workflows/pull_request.yml

```diff
@@ -0,0 +1,31 @@
+# docs: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions
+
+# This is the top-level workflow to run on all pull-requests, and invokes
+# several other workflow dependencies
+name: Pull Request
+
+on:
+  workflow_dispatch:
+  pull_request:
+
+jobs:
+  # Call auxiliary workflows
+  pre_run:
+    uses: ./.github/workflows/pre-run-check.yml
+  verilator:
+    needs: pre_run
```

Comment:
1. Why do we want to prevent the verilator tests from running if they don't have the magic hash/timestamp? Don't we want to give a drive-by contributors an early warning  if their modification doesn't even pass the verilator tests? Is this pre-check actually necessary?

Reply:
Fair analysis - there's no specific reason to gate Verilator on the pre-run check, I can make them all run from time 0.
Validating the hash/timestamp is just a way to give feedback to make sure internal pipelines are run on the code before it gets merged. Keeping honest folks honest, if you will, and giving a record of internal sign-off.

Reply:
Yeah, I understand the larger reason for having the check block merging, I just don't think it needs to be a pre-check for these other checks.